src/controllers/mainController.ts

```diff
@@ -1000,32 +978,37 @@ export default class MainController implements vscode.Disposable {
 
         // Copy object name command
         this._context.subscriptions.push(
-            vscode.commands.registerCommand(Constants.cmdCopyObjectName, async () => {
-                let node = this._objectExplorerProvider.currentNode;
-                // Folder node
-                if (node.context.type === Constants.folderLabel) {
-                    return;
-                } else if (
-                    node.context.type === Constants.serverLabel ||
-                    node.context.type === Constants.disconnectedServerNodeType
-                ) {
-                    const label = typeof node.label === "string" ? node.label : node.label.label;
-                    await this._vscodeWrapper.clipboardWriteText(label);
-                } else {
-                    let scriptingObject = this._scriptingService.getObjectFromNode(node);
-                    const escapedName = Utils.escapeClosingBrackets(scriptingObject.name);
-                    if (scriptingObject.schema) {
-                        let database = ObjectExplorerUtils.getDatabaseName(node);
-                        const databaseName = Utils.escapeClosingBrackets(database);
-                        const escapedSchema = Utils.escapeClosingBrackets(scriptingObject.schema);
-                        await this._vscodeWrapper.clipboardWriteText(
-                            `[${databaseName}].${escapedSchema}.[${escapedName}]`,
-                        );
+            vscode.commands.registerCommand(
+                Constants.cmdCopyObjectName,
+                async (node: TreeNodeInfo) => {
+                    // Folder node
+                    if (node.context.type === Constants.folderLabel) {
+                        return;
+                    } else if (
+                        node.context.type === Constants.serverLabel ||
+                        node.context.type === Constants.disconnectedServerNodeType
+                    ) {
+                        const label =
+                            typeof node.label === "string" ? node.label : node.label.label;
+                        await this._vscodeWrapper.clipboardWriteText(label);
                     } else {
-                        await this._vscodeWrapper.clipboardWriteText(`[${escapedName}]`);
+                        let scriptingObject = this._scriptingService.getObjectFromNode(node);
+                        const escapedName = Utils.escapeClosingBrackets(scriptingObject.name);
+                        if (scriptingObject.schema) {
+                            let database = ObjectExplorerUtils.getDatabaseName(node);
+                            const databaseName = Utils.escapeClosingBrackets(database);
+                            const escapedSchema = Utils.escapeClosingBrackets(
+                                scriptingObject.schema,
+                            );
+                            await this._vscodeWrapper.clipboardWriteText(
```

Comment:
All of database, schema, and escapedName are created using `escapeClosingBrackets()`, but databaseName and escapedName get brackets in the final string and escapedSchema doesn't?

Reply:
We already had a method used in the OE drag-and-drop handler to generate a qualified name for a node. While it’s not perfect, it matches what SSMS uses. I’ve updated the code to use that method here as well, so the behavior is now consistent between drag-and-drop and “Copy Object Name,” and also aligns across our products.